src/main/scala/au/com/agiledigital/toolform/model/Project.scala

```diff
@@ -0,0 +1,69 @@
+package au.com.agiledigital.toolform.model
+
+import pureconfig.{ConfigConvert, ConfigFieldMapping, ProductHint}
+import pureconfig.ConfigConvert.viaStringOpt
+
+/**
+  * Configuration for a project.
+  *
+  * @param id         the id of the project.
+  * @param name       the name of the project.
+  * @param components the components that will be built and combined to form the projects.
+  * @param resources  the resources that must be supplied to the project to run.
+  * @param topology   the network topology of the environment explains how elements will be related/connected.
+  * @param volumes    the volumes that will be mounted into components.
+  */
+case class Project(id: String,
+                   name: String,
+                   components: Map[String, Component],
+                   resources: Map[String, Resource],
+                   topology: Topology,
+                   volumes: Option[Seq[Volume]],
+                   componentGroups: Option[Seq[ComponentGroup]])
+
+/**
+  * the network topology of the environment explains how elements will be related/connected.
+  * @param links the links between components, other components and resources.
+  * @param edges the edges that will make the components and resources available outside the project.
+  */
+case class Topology(links: Seq[Link], edges: Map[String, Edge])
+
+object Topology {
+  implicit val fieldMapping: ProductHint[Topology] = ProductHint[Topology](new ConfigFieldMapping {
+    def apply(fieldName: String): String = fieldName
```

Comment:
This `fieldMapping` looks like it does nothing. Could you add a comment explaining why it is needed?